Valeriia/Movies_database_project/src/App.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { Component } from "react";
+import Header from "./components/Header";
+import Footer from "./components/Footer";
+import Movies from "./components/Movies";
+import SortPannel from "./components/SortPannel";
+import Loader from "./components/Loader";
+import movies from "./movies.json";
+import NotFound from "./components/NotFound";
+import Movie from "./components/Movie";
+import { IMovie } from "./types";
+import "./App.scss";
+
+interface IAppState {
+  movies: IMovie[] | [];
+  isLoading: boolean;
+  tempListOfMovies: IMovie[] | [];
+  filterBy: string;
+  sortBy: string;
+  searchTerm: string;
+  movie?: IMovie;
+}
+
+class App extends Component<{}, IAppState> {
+  state = {
+    movies: [],
+    isLoading: false,
+    tempListOfMovies: [],
+    filterBy: "title",
```

Comment:
Move filterBy and searchTerm into the separate component's state and apply those changes only after user clicks Search button. Also, it would be logical if search input filed get cleared after user changes searchBy